### PR TITLE
Convert app index pages to use yaml

### PR DESCRIPTION
**Pull request by pmahnke**
## Done

- Convert app index pages to use yaml and markdown templates
- I added all the text and images to the index.md files and all the logic to the _base_appliances_index.html

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8001/appliance/openhab, http://0.0.0.0:8001/appliance/plex or http://0.0.0.0:8001/appliance/adguard
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)

### CODE REVIEW

**Comment by anthonydillon** [comment]
@pmahnke are you refactoring this one?

**Comment by pmahnke** [comment]
@anthonydillon I refactored as @nottrobin and I discussed, I still think we can do more to simplify the download pages as they are all identical

File: templates/appliance/adguard/index.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+wrapper_template: "appliance/shared/_base_appliance_index.html"
+context:
+  name: "Adguard"
+  short_name: "adguard"
+  appliance_name: "Adguard"
+  company_name: "AdGuard, (ameshkov)"
+  logo: "https://assets.ubuntu.com/v1/1ec45573-AdGuard.png?w=90"
```
Comment (anthonydillon) [inline templates/appliance/adguard/index.md:8]: The width is applied in the base template so can be removed here.

File: templates/appliance/adguard/intel-nuc.html [hunk 2]
```diff
@@ -10,6 +10,6 @@
 
 {% include 'appliance/shared/_intel-nuc.html' %}
 
-{% include 'appliance/adguard/_start-using.html' %}
+{% include 'appliance/'+appliance.url_name+'/_next-steps.html' %}
```
Comment (anthonydillon) [inline templates/appliance/adguard/intel-nuc.html:13]: We might want to remove the _ on templates to sace issues in the future.

File: templates/appliance/openhab/index.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+---
+wrapper_template: "appliance/shared/_base_appliance_index.html"
+context:
+  name: "OpenHAB"
+  short_name: "openhab"
+  appliance_name: "OpenHAB"
+  company_name: "openHAB Foundation e.V. (openhab)"
+  logo: "https://assets.ubuntu.com/v1/f4632e06-openhab-logo.png?w=90"
```
Comment (anthonydillon) [inline templates/appliance/openhab/index.md:8]: Strip the `?w=90` please

File: templates/appliance/adguard/intel-nuc.html (hunk 2 above)
Comment (nottrobin) [inline templates/appliance/adguard/intel-nuc.html:13]: I wonder if f-strings work in templates?

```suggestion
{% include f"appliance/{appliance.url_name}/_next-steps.html" %}
```

File: appliances.yaml [hunk 4]
```diff
@@ -1,6 +1,7 @@
 appliances:
   plex:
     name: "Plex"
+    url_name: "plex"
```
Comment (nottrobin) [inline appliances.yaml:4]: Why do you still need this file? Is this used by the index page or something?

File: templates/appliance/shared/_base_appliance_index.html [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        <iframe class="u-embedded-media__element" src="{{ screenshots.1 }}" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+        {% else %}
+        <img src="{{ screenshots.1 }}" class="p-image--shadowed" alt="{{ appliance_name }} screenshots">
+        {% endif %}
+     </div>
+    </div>
+    <div class="col-2 u-align--center">
+      <ul class="p-list row">
+        {% for image in screenshots.values() %}
+          {% if "youtu" not in image %}
```
Comment (anthonydillon) [inline templates/appliance/shared/_base_appliance_index.html:65]: Remove indentation please

File: templates/appliance/adguard/intel-nuc.html (hunk 2 above)
Comment (nottrobin) [reply templates/appliance/adguard/intel-nuc.html:13]: @anthonydillon the idea is that we distinguish explicitly between include files that are internal to the template system and publicly exposed template files. This is to stop people just inspecting our template files through the website. This `_next-steps.html` file is an internal include file right? So it should explicitly *not* have the underscore removed.

What we've done with `navigation-h ...` files is pretty hacky, but you have to think about it as us deliberately creating a public page containing some HTML for JavaScript to consume, not as a template include file, which it's not.

Comment (nottrobin) [reply templates/appliance/adguard/intel-nuc.html:13]: This also follows the pattern used in Jekyll and Eleventy and other static site systems, where `_` files are removed, and includes are kept in an `_includes` folder. I think this is much neater and what we should try to replicate.

Comment (pmahnke) [reply templates/appliance/adguard/intel-nuc.html:13]: not to f strings; however, it does seem to like `~` to do the same - https://jinja.palletsprojects.com/en/2.11.x/templates/

File: appliances.yaml (hunk 4 above)
Comment (pmahnke) [reply appliances.yaml:4]: it is used on the download pages... unless it is available some other way?

**Pull request merged**